### DISCUSSION TITLE
How to disable password in heroku?

### DISCUSSION THREAD

**Discussion by PakkiPavan** [coder/code-server#3821] [Q&A] [answered]
Is it possible to bypass password in code-server which is deployed in heroku?
I have deployed code-server in heroku using steps mentioned in below URL:
https://github.com/cdr/deploy-code-server/blob/main/guides/heroku.md
I want to disable password when logging in code-server. How to achieve that?

**Comment by ghost** [comment]
Remove `PASSWORD` and `HASHED_PASSWORD` env vars if present https://github.com/cdr/deploy-code-server/tree/main/deploy-container#environment-variables. Since anyone can view heroku deployments, its much more safe to keep atleast one form of protection.

**Reply by PakkiPavan** [reply]
I have removed PASSWORD env variable in heroku. Still it is asking for password.
Any other place these env vars available?

**Reply by ghost** [reply]
Im not too sure about heroku but was your instance re-deployed? If not, then you could trigger a rebuild for the changes to take effect.

**Reply by jsjoeio** [reply]
Hmm... cc @bpmct who is the Heroku wizard. 

There are two ways you can disable the password:
- `code-server --auth none` 
- `auth: none` in your `config.yaml`

I'm not sure how we can do that on Heroku but Ben might know

**Reply by PakkiPavan** [reply]
> Im not too sure about heroku but was your instance re-deployed? If not, then you could trigger a rebuild for the changes to take effect.


Yes, I have re-deployed, still it is asking for password.

**Comment by bpmct** [comment, marked as answer]
Hey @PakkiPavan if there is no PASSWORD env var, a random one is created. You will need to set auth: none in config.yaml or launch it with --auth none.

In the Dockerfile you can copy a code-server config file OR add the flag in the entrypoint of your repo. Then, re deploy.

**Reply by PakkiPavan** [reply]
Now able to launch code-server without password after adding flag --auth none in the entrypoint.
Thanks @bpmct for the response.
